sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1462,13 +1480,50 @@ export class BlobClient extends StorageClient {
    */
   public async downloadToBuffer(
     buffer: Buffer,
-    offset: number,
+    offset?: number,
     count?: number,
-    options: BlobDownloadToBufferOptions = {}
-  ): Promise<Buffer> {
+    options?: BlobDownloadToBufferOptions
+  ): Promise<Buffer>;
+
+  public async downloadToBuffer(
+    param1: Buffer | number = 0,
+    param2: number = 0,
+    param3: BlobDownloadToBufferOptions | number = 0,
+    param4: BlobDownloadToBufferOptions = {}
+  ) {
+    let buffer: Buffer;
+    let offset: number;
+    let count: number;
+    let options = param1 instanceof Buffer ? param4 : (param3 as BlobDownloadToBufferOptions);
```

Comment:
In the case of `downloadToBuffer()`, `options` will get set to 0
I would suggest not setting any default values in the method signature when working with overloads and if the parameter is a union type to keep things simple.

My suggestion would be
```javascript
  public async downloadToBuffer(
    param1?: Buffer | number,
    param2?: number,
    param3?: number | BlobDownloadToBufferOptions,
    param4: BlobDownloadToBufferOptions = {}
  ): Promise<Buffer> {
    let buffer: Buffer | undefined;
    let offset = 0;
    let count = 0;
    let options = param4;

    if (param1 instanceof Buffer) {
      buffer = param1;
      offset = param2 || 0;
      count = typeof param3 === "number" ? param3 : 0;
    } else {
      offset = typeof param1 === "number" ? param1 : 0;
      count = typeof param2 === "number" ? param2 : 0;
      options = param3 as BlobDownloadToBufferOptions;
    }
    if (!count) {
      // call getProperties to figure out value for count
    }
    if (!buffer) {
      // allocate buffer
    }
```

Reply:
Updated.